apps/dashboard/src/components/pay/PayAnalytics/components/TotalPayVolume.tsx

```diff
@@ -1,159 +1,82 @@
-  projectId: string;
-  teamId: string;
-  from: Date;
-  to: Date;
-  numberOfDays: number;
-}) {
-  const [intervalType, setIntervalType] = useState<"day" | "week">(
-    props.numberOfDays > 30 ? "week" : "day",
-  );
-
-  // if prop changes, update intervalType
-  // eslint-disable-next-line no-restricted-syntax
-  useEffect(() => {
-    setIntervalType(props.numberOfDays > 30 ? "week" : "day");
-  }, [props.numberOfDays]);
-
-  const volumeQuery = processQuery(
-    usePayVolume({
-      /**
-       *  @deprecated - remove after migration
-       */
-      clientId: props.clientId,
-      // switching to projectId for lookup, but have to send both during migration
-      projectId: props.projectId,
-      teamId: props.teamId,
-      from: props.from,
-      intervalType,
-      to: props.to,
-    }),
-  );
-
-  return (
-    <section className="relative flex flex-col justify-center">
-      {!volumeQuery.isError ? (
-        <RenderData
-          query={volumeQuery}
-          intervalType={intervalType}
-          setIntervalType={setIntervalType}
-        />
-      ) : (
-        <FailedToLoad />
-      )}
-    </section>
-  );
-}
-
-function RenderData(props: {
-  query: ProcessedQuery;
-  intervalType: "day" | "week";
-  setIntervalType: (intervalType: "day" | "week") => void;
+  data: UniversalBridgeStats[];
+  dateFormat?: {
+    month: "short" | "long";
+    day?: "numeric" | "2-digit";
+  };
 }) {
   const uniqueId = useId();
   const [successType, setSuccessType] = useState<"success" | "fail">("success");
   const [type, setType] = useState<"all" | "crypto" | "fiat">("all");
 
-  const graphData: GraphData[] | undefined =
-    props.query.data?.intervalResults.map((x) => {
-      const date = format(new Date(x.interval), "LLL dd");
-
+  const graphData: GraphData[] | undefined = useMemo(() => {
+    let data = (() => {
       switch (type) {
         case "crypto": {
-          return {
-            date,
-            value:
-              x.buyWithCrypto[
-                successType === "success" ? "succeeded" : "failed"
-              ].amountUSDCents / 100,
-          };
+          return props.data?.filter((x) => x.type === "onchain");
         }
-
         case "fiat": {
-          return {
-            date,
-            value:
-              x.buyWithFiat[successType === "success" ? "succeeded" : "failed"]
-                .amountUSDCents / 100,
-          };
+          return props.data?.filter((x) => x.type === "onramp");
         }
-
         case "all": {
-          return {
-            date,
-            value:
-              x.sum[successType === "success" ? "succeeded" : "failed"]
-                .amountUSDCents / 100,
-          };
+          return props.data;
         }
-
         default: {
           throw new Error("Invalid tab");
         }
       }
-    });
+    })();
+
+    data = (() => {
+      if (successType === "fail") {
+        return data.filter((x) => x.status === "failed");
```

Comment:
just a nit: noticed that on an empty dashboard, it shows the bar fully red, with 0 success 0 failed